### PR TITLE
Add a rough outline/draft

### PR DESCRIPTION
**Pull request by estark37**
Here's a basic outline, modeled on the HPKP spec, with some text and notes filled in.

### CODE REVIEW

File: draft-stark-expect-ct.md [hunk 1]
```diff
@@ -23,6 +23,316 @@ TODO: Wite an abstract here
...
+### The report-uri Directive
+
+The OPTIONAL `report-uri` directive indicates the URI to which the UA SHOULD
+report Expect-CT failures ({{expect-ct-compliance}}). The UA POSTs the reports
+to the given URI as described in {{reporting-expect-ct-failure}}.
+
+Hosts may set `report-uri`s that use HTTP or HTTPS. If the scheme in the
+`report-uri` is one that uses TLS (e.g., HTTPS), UAs MUST check Expect-CT
+compliance when the host in the `report-uri` is an Expect-CT host; similarly,
+UAs MUST apply HSTS if the host in the `report-uri` is a Known HSTS Host.
```
Comment (jcjones) [inline draft-stark-expect-ct.md:95]: We may get in trouble about making a normative requirement on HSTS in this document.

File: draft-stark-expect-ct.md [hunk 2]
```diff
@@ -23,6 +23,316 @@ TODO: Wite an abstract here
...
+
+# Interactions with Preloaded Pin Lists
+
+# Reporting Expect-CT Failure
+
+When the UA receives an Expect-CT header with a `report-uri` directive that does
+not comply with the UA's CT policy, or when the UA connects to a noted Expect-CT
+host that does not comply with the CT policy, the UA SHOULD report Expect-CT
+failures to the configured `report-uri`. The UA does this by POSTing a JSON
+{{!RFC7159}} message to the URI. The JSON message takes this form:
```
Comment (jcjones) [inline draft-stark-expect-ct.md:308]: Recommend something like this... `webappapis#*` pointing to http://www.w3.org/TR/html5/webappapis.html and `infrastructure#fetch` to http://www.w3.org/TR/html5/infrastructure.html#fetch

> 1. Prepare a JSON object report object with a single key, `expect-ct-report`, whose value is the result of generating a violation report object. 
> 2. Let `report body` be the JSON stringification of report object. 
> 3. {{webappapis#Queue a task}} to {{infrastructure#fetch}} `report-uri` from the origin of the protected resource, with the synchronous flag _not_ set, using HTTP method `POST`, with a `Content-Type` header field of `application/expect-ct-report`, and an entity body consisting of `report body`. If the origin of `report URL` is not the same as the origin of the protected resource, the block cookies flag MUST also be set. The user agent MUST NOT follow redirects when fetching this resource. (Note: The user agent ignores the fetched resource.) The {{webappapis#task source}} for these {{webappapis#concept-tasks}} is the **Content Security Policy task source**.

File: draft-stark-expect-ct.md [hunk 3]
```diff
@@ -23,6 +23,316 @@ TODO: Wite an abstract here
...
+EXPLAIN report fields here
+
 # Security Considerations
 
+## Maximum max-age
+
+1 year?
+
+# Privacy Considerations
+
```
Comment (jcjones) [inline draft-stark-expect-ct.md:328]: If JS attempts an async load of a HTTPS resource that has expect-ct set, you can determine if the evaluating user agent has been to that site recently.

If there's a `report-uri`, at least the report-receiver could notice this behavior, though.

File: draft-stark-expect-ct.md (hunk 1 above)
Comment (estark37) [reply draft-stark-expect-ct.md:95]: I copied this language from the HPKP spec: https://tools.ietf.org/html/rfc7469#section-2.1.4

I agree it's a little weird though

**Pull request merged**